File: src/pnp/compiler/model/type/AbstractType.java
package pnp.compiler.model.type;

import pnp.compiler.model.Variable;

import java.util.List;

public class AbstractType implements Type {
    final private List<Variable> attributes;
    final String name;

    public AbstractType(String name, List<Variable> attributes) {
        this.attributes = attributes;
        this.name = name;
    }

    @Override
    public boolean isTypeOf(Object value) {
        if (value instanceof Variable) {
            return ((Variable) value).getType().equals(this);
        }
        return false;
    }
}

File: src/main/kotlin/br/com/pnp/model/construct/type/AbstractType.kt
package br.com.pnp.model.construct.type

import br.com.pnp.model.construct.Construct
import br.com.pnp.model.construct.Variable

class AbstractType(private val name: String, private val attributes: List<Variable>): Type(), Construct {
    override fun isTypeOf(value: Any): Boolean {
        TODO ("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun toString(): String {
        return name
    }
}